common/src/main/java/com/wynntils/models/raid/RaidModel.java

```diff
@@ -94,10 +95,14 @@ public void onTitle(TitleSetTextEvent event) {
     @SubscribeEvent
     public void onChatMessage(ChatMessageReceivedEvent event) {
         if (inBuffRoom()) {
-            WynntilsMod.info(event.getStyledText().getString());
-            Matcher matcher = event.getStyledText().getMatcher(RAID_CHOOSE_BUFF_PATTERN);
+            Matcher matcher = event.getOriginalStyledText().stripAlignment().getMatcher(RAID_CHOOSE_BUFF_PATTERN);
             if (matcher.matches()) {
                 String playerName = matcher.group(2);
+                // if the player is nicknamed
+                if (playerName.startsWith("§o")) {
+                    playerName = RevealNicknamesFeature.getNameAndNick(event).a();
```

Comment:
This is not the right way to do this, we don't like features referencing each other like this.

I would suggest either add a new method to StyledTextUtils that will extract just the username or simply moving the `getNameAndNick` method to StyledTextUtils

Reply:
I will refactor this as soon as I'm home later today, I had a look at the architecture wiki page and couldn't see anything about not exposing parts of features publicly. Maybe I missed it though, a util function is perfect either way.

Reply:
I don't know if the wiki is up to date, a lot has changed since it was originally made but the idea is that a feature should only rely on itself, any relevant Models/Services/Managers/Handlers and other util classes. It should never rely on something contained in another feature.

If 2 features are doing the same, or similar things like in this case, then it's best to move that logic somewhere else

Reply:
I don't think Wiki ever changed well except for addition of functions at some point, but the wiki is the same as it was when I joined which was during 1.18 days (there wasn't even a config screen at the time)

Reply:
Other than that. `playerName.startsWith("§o")` can be solved by an optional group in the regex, and a separate group for an unformatted name string.